### ISSUE TITLE
Fatal failure at CVC4/src/./theory/bv/bitblast/bitblast_strategies_template.h:46

### ISSUE THREAD

**Issue by muchang** [CVC4/CVC4#3625]
Hi,
For this formula,
```
(assert (exists ((a (_ BitVec 32)))
  (forall ((b (_ BitVec 32)))
    (exists ((c (_ BitVec 32)))
      (forall ((d (_ BitVec 32)))
        (or (and (= b c) (= c #x00000004) (distinct d (bvadd c #x00000001)))
            (bvslt (bvmul ((_ sign_extend 33) a) ((_ zero_extend 33) d))
                   (bvmul ((_ sign_extend 33) a) ((_ zero_extend 33) b)))))))))
(check-sat)
```
CVC4 throws out a fatal failure:
```
Fatal failure within T CVC4::theory::bv::UndefinedAtomBBStrategy(CVC4::TNode, CVC4::theory::bv::TBitblaster<T>*) [with T = CVC4::NodeTemplate<true>; CVC4::TNode = CVC4::NodeTemplate<false>] at CVC4/src/./theory/bv/bitblast/bitblast_strategies_template.h:46
Unreachable code reached
Aborted
```

OS: Ubuntu 18.04
Revision: c3bc4ac (assertion on)

**Comment by nafur** [comment]
This issue persists. This example triggers a bit faster:
```
(assert
    (exists
        ((a (_ BitVec 1)))
        (forall
            ((b (_ BitVec 1)))
            (exists
                ((c (_ BitVec 1)))
                (forall
                    ((d (_ BitVec 1)))
                    (or
                        (and
                            (= b c)
                            (= c (_ bv0 1))
                            (distinct d (bvadd c (_ bv1 1)))
                        )
                        (bvslt
                            (bvmul ((_ sign_extend 1) a) ((_ zero_extend 1) d))
                            (bvmul ((_ sign_extend 1) a) ((_ zero_extend 1) b))
                        )
                    )
                )
            )
        )
    )
)
(check-sat)
```

**Comment by nafur** [comment]
This issue was fixed with a35585fc3b20d70f88a156cd0403f6aa5c9a0dbe.
